pkg/v0/cmd.go

```diff
@@ -224,19 +305,33 @@ func detectNewCommits(ctx context.Context, logger *logrus.Entry, stagingDir, cen
 		case flags.HTTPS:
 			remote = "https://github.com/operator-framework/" + repo + ".git"
 		}
-		if _, err := internal.RunCommand(repoLogger, exec.CommandContext(ctx,
+		ref, ok := repoRefs["operator-framework/" + repo]
+		if !ok {
+			logger.Debugf("using 'master' for repo %q", repo)
+			ref = "master"
+		} else {
+			logger.Debugf("using %q for repo %q", ref, repo)
+		}
+		output, err := internal.RunCommand(repoLogger, exec.CommandContext(ctx,
 			"git", "fetch",
+			"--porcelain",
 			remote,
-			"master",
-		)); err != nil {
+			ref,
+		))
+		if err != nil {
 			return nil, err
 		}
+		strs := strings.Split(output, " ")
+		if len(strs) != 4 {
+			return nil, fmt.Errorf("unexpected command output %q", output)
+		}
+		fetchRef := strs[2]
```

Comment:
Why? The _whole point_ of `FETCH_HEAD` is to refer to what you just fetched. This adds more logic to the script and the best you can do is correctly parse the `--porcelain` output to get what `FETCH_HEAD` already refers to. Please don't do this.

Reply:
I could be mis-remembering, but I recall there was a situation where `git fetch` didn't actually fetch anything, so `FETCH_HEAD` wasn't updated. AFAIR, this was not as much an issue running the bumper as a job, but running the bumper locally.

Reply:
After I cleared out the old remotes that the old scripts had used, it seems to be better behaved without this.